test/toNumber.js

```diff
@@ -0,0 +1,93 @@
+import { assert } from 'chai';
+import * as R from 'ramda';
+
+import BigInt from './shared/BigInt';
+import * as RA from '../src';
+
+describe('toNumber', function () {
+  context(
+    'given a [non NaN] number or a number-convertible value',
+    function () {
+      specify('should return a [non NaN] number', function () {
```

Comment:
I'm not sure I agree with this system of assertions.

What would you say to this system ?

Before
```js
        result = RA.toNumber(12e3);
        assert.isNumber(result);
        assert.isNotNaN(result);

        result = RA.toNumber(Infinity);
        assert.isNumber(result);
        assert.isNotNaN(result);

        result = RA.toNumber(null);
        assert.isNumber(result);
        assert.isNotNaN(result);

        result = RA.toNumber(undefined);
        assert.isNaN(result);
```
After
```js
        assert.strictEqual(RA.toNumber(12e3), 12e3);
        assert.strictEqual(RA.toNumber(Infinity), Infinity);        
        assert.strictEqual(RA.toNumber(null), 0);
        assert.isNaN(RA.toNumber(undefined));
```

It's better to check to real values and explicitly see them tests. From your assertions it's not that apparent what the resulting value should be.